src/components/ReportActionItem/ReportPreview.js

```diff
@@ -0,0 +1,159 @@
+    }),
+
+    /** Session info for the currently logged in user. */
+    session: PropTypes.shape({
+        /** Currently logged in user email */
+        email: PropTypes.string,
+    }),
+
+    /** Popover context menu anchor, used for showing context menu */
+    contextMenuAnchor: PropTypes.shape({current: PropTypes.elementType}),
+
+    /** Callback invoked when View Details is pressed */
+    onViewDetailsPressed: PropTypes.func,
+
+    /** Callback for updating context menu active state, used for showing context menu */
+    checkIfContextMenuActive: PropTypes.func,
+
+    /** Whether the IOU is hovered so we can modify its style */
+    isHovered: PropTypes.bool,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    contextMenuAnchor: null,
+    isHovered: false,
+    iouReport: {},
+    onViewDetailsPressed: () => {},
+    checkIfContextMenuActive: () => {},
+    session: {
+        email: null,
+    },
+};
+
+const ReportPreview = (props) => {
+    const reportAmount = CurrencyUtils.convertToDisplayString(props.iouReport.total, props.iouReport.currency);
+    const managerEmail = props.iouReport.managerEmail || '';
+    const managerName = ReportUtils.getDisplayNameForParticipant(managerEmail, true);
+    const isCurrentUserManager = managerEmail === lodashGet(props.session, 'email', null);
+    return (
+        <View style={[styles.chatItemMessage, styles.mt4]}>
+            {_.map(props.action.message, (index) => (
+                <Pressable
+                    key={`ReportPreview-${props.action.reportActionID}-${index}`}
+                    onPress={props.onViewDetailsPressed}
+                    onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
+                    onPressOut={() => ControlSelection.unblock()}
+                    onLongPress={(event) => showContextMenuForReport(event, props.contextMenuAnchor, props.chatReportID, props.action, props.checkIfContextMenuActive)}
+                    style={[styles.flexRow, styles.justifyContentBetween]}
+                    focusable
+                >
+                    <View>
+                        {props.iouReport.hasOutstandingIOU ? (
+                            <Text style={[styles.chatItemMessage, styles.cursorPointer]}>{props.translate('iou.payerOwesAmount', {payer: managerName, amount: reportAmount})}</Text>
+                        ) : (
+                            <View style={[styles.flexRow]}>
+                                <Text style={[styles.chatItemMessage, styles.cursorPointer]}>{props.translate('iou.payerSettled', {amount: reportAmount})}</Text>
+                                {!props.iouReport.hasOutstandingIOU && (
+                                    <Icon
+                                        style={[styles.ml10]}
+                                        src={Expensicons.Checkmark}
+                                        fill={themeColors.iconSuccessFill}
+                                    />
```

Comment:
This change caused a regression [here](https://github.com/Expensify/App/issues/19049).